app/Services/Registrar.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Gladiator\Services;
+
+use Gladiator\Models\User;
+use Gladiator\Services\Northstar\Northstar;
+use Gladiator\Services\Northstar\Exceptions\NorthstarUserNotFoundException;
+
+class Registrar
+{
+    /**
+     * Northstar instance.
+     *
+     * @var \Gladiator\Services\Northstar\Northstar
+     */
+    protected $northstar;
+
+    /**
+     * Create new Registrar instance.
+     *
+     * @param Northstar $northstar
+     */
+    public function __construct(Northstar $northstar)
+    {
+        $this->northstar = $northstar;
+    }
+
+    /**
+     * Resolve user within system or create new user.
+     *
+     * @param  array  $credentials
+     * @return \Gladiator\Models\User
+     */
+    public function findOrCreate($credentials)
+    {
+        $account = $this->hasAccountInSystem($credentials['type'], $credentials['key']);
+
+        if ($account instanceof User) {
+            $user = $account;
+
+            session()->flash('status', 'User already exists!');
```

Comment:
Yeah, I had a bit of a conflict deciding on how to resolve this... So you're saying then in the respective controllers (web vs api) it handles how to respond from there, but calls 2 methods instead of 1? So the check for `instanceof User` gets repeated in both?